backend/api/profile.py

```diff
@@ -24,6 +38,15 @@ def update_profile(
     pid_onyen: tuple[int, str] = Depends(authenticated_pid),
     user_svc: UserService = Depends()
 ):
+    """Update a user's profile. If the user does not exist, create a new user.
+
+    Since the user is authenticated, we can trust the pid and onyen. However,
+    since the user may not be registered, we depend on authenticated_pid rather
+    than registered_user.
+
+    ProfileForm is used here, rather than User, for similar registration-specific
+    purposes. Importantly, it does not include the ID.
```

Comment:
This perhaps could be phrased a bit better. 

I got it, but I can see some confusion from the wording "the ID." 

Maybe something along the lines of: "Importantly, ProfileForm doesn't contain an ID field"?